pyhive/tests/test_hive.py

```diff
@@ -204,14 +202,35 @@ def test_custom_transport(self):
         socket = thrift.transport.TSocket.TSocket('localhost', 10000)
         sasl_auth = 'PLAIN'
 
-        def sasl_factory():
+        spec_sasl = importlib.util.find_spec('sasl')
+        spec_puresasl = importlib.util.find_spec('puresasl')
+
+        def get_sasl_client():
+            import sasl
             sasl_client = sasl.Client()
             sasl_client.setAttr('host', 'localhost')
             sasl_client.setAttr('username', 'test_username')
             sasl_client.setAttr('password', 'x')
             sasl_client.init()
             return sasl_client
+        
+        def get_pure_sasl_client():
+            from pyhive.sasl_compat import PureSASLClient
+            sasl_client = PureSASLClient(host='localhost', username='test_username', password='x')
+            return sasl_client
+
+        def sasl_factory():
```

Comment:
can we reuse another sasl_factory method here that's defined in hive.py ?

Reply:
done